corehq/apps/app_manager/util.py

```diff
@@ -409,7 +409,7 @@ def get_casedb_schema(form):
     This lists all case types and their properties for the given app.
     """
     app = form.get_app()
-    base_case_type = form.get_module().case_type
+    base_case_type = form.get_module().case_type if form.requires_case() else None
```

Comment:
As for reverting, it may unfortunately be worth it. Timings for the vellum schema URL (my change was merged Apr. 19): https://app.datadoghq.com/metric/explorer?live=true&page=0&is_auto=false&from_ts=1490650456272&to_ts=1493242456272&tile_size=m&exp_metric=nginx.timings&exp_scope=url%3A%2Fa%2F_%2Fapps%2Fschema%2Fform%2F_%2F&exp_agg=max&exp_row_type=metric.